promql/engine.go

```diff
@@ -1074,11 +1074,44 @@ func (ev *evaluator) eval(expr Expr) Value {
 				putPointSlice(ss.Points)
 			}
 		}
+
+		putPointSlice(points)
+
+		// The absent_over_time function returns 0 or 1 Series. So far, the Matrix
+		// contains multiple series. The following code will create a new Series
+		// with values of 1 for the timestamps where no Series has value.
+		if e.Func.Name == "absent_over_time" {
+			found := map[int64]struct{}{}
```

Comment:
@brian-brazil should I move this block code in functions.go?

return absentOverTimePostProcessing(mat)

Reply:
I see no need to move it over, it's part of the engine itself.